Adaptors/Amqp/src/ConnectionAmqp.cs

```diff
@@ -69,18 +68,43 @@ public Task<HealthCheckResult> Check(HealthCheckTag tag)
   public async Task Init(CancellationToken cancellationToken = default)
     => await connectionTask_;
 
-  private static async Task InitTask(ConnectionAmqp    conn,
-                                     CancellationToken cancellationToken = default)
+  public async Task<Connection> GetConnectionAsync(CancellationToken cancellationToken = default)
+  {
+    if (connection_ is null || connection_.IsClosed)
+    {
+      connection_ = await CreateConnection(options_,
```

Comment:
Add an executionSingleizer to prevent multiple connections to be created by multiple threads.